substrate/client/consensus/babe/src/lib.rs

```diff
@@ -1413,6 +1371,70 @@ where
 		let hash = block.post_hash();
 		let number = *block.header.number();
 		let info = self.client.info();
+		let parent_hash = *block.header.parent_hash();
+
+		let create_inherent_data_providers = self
+			.create_inherent_data_providers
+			.create_inherent_data_providers(parent_hash, ())
+			.await?;
+
+		let slot_now = create_inherent_data_providers.slot();
+
+		let babe_pre_digest = find_pre_digest::<Block>(&block.header)
+			.map_err(|e| ConsensusError::Other(Box::new(e)))?;
+		let slot = babe_pre_digest.slot();
+
+		// Check inherents.
+		if let Some(inner_body) = block.body {
+			let new_block = Block::new(block.header.clone(), inner_body);
+			if !block.state_action.skip_execution_checks() {
```

Comment:
This should wrap the all the things that are required to check the inherents.

Also as we don't use the inherente data providers etc for anything else in here, we can just move the code directly to `check_inherents`.